### PR TITLE
Bug 1927902: i18n Cluster Utilization dashboard durations

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Durations (i.e. 1 hour, 24 hours, etc.) were not internationalized in dashboard metric cards. I internationalized them so we can get those translated. 

TranslatedDuration also wasn't being used anywhere, so I refactored it so we could use it.

I'm passing the TFunction into a couple of components because I couldn't see a way around it - using i18next.t triggered missingKey errors and there were too many components using them to nest them inside a parent component. 

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1927902.

### CODE REVIEW

File: frontend/packages/console-shared/src/components/dashboard/duration-hook.ts [hunk 1]
```diff
@@ -1,48 +1,32 @@
...
+
+export const Duration = (t: TFunction) => {
+  return {
+    ONE_HR: t('console-shared~1 hour'),
+    SIX_HR: t('console-shared~6 hours'),
+    TWENTY_FOUR_HR: t('console-shared~24 hours'),
+  };
+};
 
+export const TranslatedDuration = (duration, t: TFunction) => {
```
Comment (jhadvig) [inline frontend/packages/console-shared/src/components/dashboard/duration-hook.ts:12]:
```suggestion
export const TranslatedDuration = (duration, t: TFunction): TFunction => {
```

File: frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx [hunk 2]
```diff
@@ -175,12 +176,26 @@ export const PrometheusUtilizationItem = withDashboardResources<PrometheusUtiliz
     let request: PrometheusResponse, requestError: any;
     let isLoading = false;
 
+    const { t } = useTranslation();
+
+    const UTILIZATION_QUERY_DURATION = {
+      ONE_HR: t('public~1 hour'),
```
Comment (jhadvig) [inline frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx:182]: why do we need these string both to be in `public` and `console-shared` locales? Is it OK if they are duplicate ?

File: frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx [hunk 3]
```diff
@@ -175,12 +176,26 @@ export const PrometheusUtilizationItem = withDashboardResources<PrometheusUtiliz
     let request: PrometheusResponse, requestError: any;
     let isLoading = false;
 
+    const { t } = useTranslation();
+
+    const UTILIZATION_QUERY_DURATION = {
```
Comment (jhadvig) [inline frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx:181]: cant we use the `Duration` defined in the `duration-hook.ts` ? so we dont repeat ourselves

File: frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx [hunk 4]
```diff
@@ -266,12 +281,26 @@ export const PrometheusMultilineUtilizationItem = withDashboardResources<
     namespace,
     isDisabled = false,
   }) => {
+    const { t } = useTranslation();
+
+    const UTILIZATION_QUERY_DURATION = {
```
Comment (jhadvig) [inline frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx:286]: cant we use the `Duration` defined in the `duration-hook.ts` ? so we dont repeat ourselves

**Review by jhadvig — CHANGES_REQUESTED** [review]

File: frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx (hunk 2 above)
Comment (rebeccaalpert) [reply frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx:182]: It shouldn't cause problems, but I can consolidate them into one namespace.

**Review by jhadvig — COMMENTED** [review]
@rebeccaalpert there seems to be an issue in the i18n files. Check the failing frontend test please

**Comment by jhadvig** [comment]
yep there is an issue with the i18n files
```
Merging /go/src/github.com/openshift/console/frontend/packages/ceph-storage-plugin/locales/en/dashboard.json with matching public namespace.
Conflict: Key "Not available" in /go/src/github.com/openshift/console/frontend/public/locales/en/dashboard.json already exists. Skipping merge for "Not available."
Merging /go/src/github.com/openshift/console/frontend/packages/console-shared/locales/en/dashboard.json with matching public namespace.
Conflict: Key "Not available" in /go/src/github.com/openshift/console/frontend/public/locales/en/dashboard.json already exists. Skipping merge for "Not available."
Conflict: Key "Updating" in /go/src/github.com/openshift/console/frontend/public/locales/en/dashboard.json already exists. Skipping merge for "Updating."
Merging /go/src/github.com/openshift/console/frontend/packages/console-shared/locales/en/machines.json with matching public namespace.
Merging /go/src/github.com/openshift/console/frontend/packages/console-shared/locales/en/public.json with matching public namespace.
Conflict: Key "Not available" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Not available."
Merging /go/src/github.com/openshift/console/frontend/packages/dev-console/locales/en/dashboard.json with matching public namespace.
Merging /go/src/github.com/openshift/console/frontend/packages/operator-lifecycle-manager/locales/en/public.json with matching public namespace.
Conflict: Key "Status" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Status."
Conflict: Key "Display name" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Display name."
Conflict: Key "Name" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Name."
Conflict: Key "None" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "None."
Conflict: Key "True" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "True."
Conflict: Key "False" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "False."
Conflict: Key "Save" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Save."
Conflict: Key "Delete" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Delete."
Conflict: Key "Labels" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Labels."
Conflict: Key "Create" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Create."
Conflict: Key "Cancel" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Cancel."
Conflict: Key "Kind" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Kind."
Conflict: Key "Latest version" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Latest version."
Conflict: Key "Created" in /go/src/github.com/openshift/console/frontend/public/locales/en/public.json already exists. Skipping merge for "Created."
Merging /go/src/github.com/openshift/console/frontend/packages/patternfly/locales/en/notification-drawer.json with matching public namespace.
Done in 25.04s.
i18n files are not up to date. Commit them to fix.
```

**Comment by rebeccaalpert** [comment]
Addressed feedback.

File: frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx [hunk 5]
```diff
@@ -266,12 +277,22 @@ export const PrometheusMultilineUtilizationItem = withDashboardResources<
     namespace,
     isDisabled = false,
   }) => {
+    const { t } = useTranslation();
```
Comment (jhadvig) [inline frontend/public/components/dashboard/dashboards-page/cluster-dashboard/utilization-card.tsx:280]: wondering if we shouldn't pull trig block (and the same one above) into a standalone function that would just return `UTILIZATION_QUERY_HOUR_MAP`, so we dont repeat ourselves

**Review by jhadvig — COMMENTED** [review]
@rebeccaalpert one last comment and we are ready to merge this 👍

**Comment by yanpzhan** [comment]
@rebeccaalpert Tested against the  pr, now the durations are marked for translated on cluster, but they are not translated into Chinese/Japanese/Korean when select related language display, for example:
Screenshot from 2021-03-23 14-54-57
Will translation files for different languages be update too?

**Comment by rebeccaalpert** [comment]
Hi @yanpzhan - The "1 hour," "6 hours," and "24 hours" text will get translated in the next file exchange with the translators after this merges. Translation into Chinese/Japanese/Korean won't be part of this PR.

**Review by jhadvig — APPROVED** [review]

**Pull request merged**